ldk/javascript/src/webpack/generate-banner.ts

```diff
@@ -10,11 +11,49 @@ export function generateMetadata(ldkSettings: LdkSettings): string {
   if (!ldkSettings || !ldkSettings.ldk || Object.keys(ldkSettings.ldk).length === 0) {
     throw new Error(permissionsErrorMessage);
   }
+
+  if (ldkSettings.ldk.configSchema) {
+    const ajv = new Ajv();
+    const errors: string[] = [];
+
+    Object.values(ldkSettings.ldk.configSchema.properties as LdkConfigProperties).forEach(
+      (configChild) => {
+        try {
+          ajv.compile(configChild);
+
+          const { default: defaultValue, type } = configChild;
+
+          // Only allow types supported by Loop Library form
+          if (type !== 'string' && type !== 'object') {
+            errors.push(
+              `Error: The LDK does not currently support "${type}" types in the config schema`,
+            );
+          }
+
+          // Check to make sure default value matches type because ajv won't
+          if (defaultValue && typeof defaultValue !== type) {
+            errors.push(
+              `Error: The default value "${defaultValue}" does not match the type "${type}"`,
+            );
+          }
+        } catch (error) {
+          errors.push(error as string);
+        }
+      },
+    );
+
+    if (errors.length) {
+      throw new Error(`There was an error with your LDK config:\n${errors.join('\n')}`);
+    }
+  }
```

Comment:
Turns out we're simplifying your job and supporting any valid JSON schema in the form. Should only need to grab the configSchema value, try compiling it with ajv, and throw an error if that fails
```suggestion
  if (ldkSettings.ldk.configSchema) {
    const ajv = new Ajv();

    try {
      ajv.compile(ldkSettings.ldk.configSchema);
    } catch (error) {
      throw new Error(`There was an error with your LDK config:\n${error}`);
    }
  }
```

Reply:
Sounds good! Changed this in https://github.com/open-olive/loop-development-kit/pull/519/commits/a9766d413c596be8470a4cc221532c3bfd940519